examples/config.py

```diff
@@ -0,0 +1,159 @@
+#
+# Copyright(c)
+#       2015-2018 George Washington University
+#       2015-2018 University of California Riverside
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# * Redistributions of source code must retain the above copyright
+#   notice, this list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright
+#   notice, this list of conditions and the following disclaimer in
+#   the documentation and/or other materials provided with the
+#   distribution.
+# * The name of the author may not be used to endorse or promote
+#   products derived from this software without specific prior
+#   written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""This script allows a user to launch a chain of NFs from a JSON
+config file passed in as an argument when running the script"""
+
+import sys
+import json
+import os
+import shlex
+import time
+from signal import signal, SIGINT
+from subprocess import Popen
+from datetime import datetime
+
+def handler(signal_received, frame):
+    """Handles cleanup on user shutdown"""
+    # pylint: disable=unused-argument
+    for lf in log_files:
+        lf.close()
+    print("\nExiting...")
+    sys.exit(0)
+
+def get_config():
+    """Gets path to JSON config file"""
+    if len(sys.argv) < 2:
+        print("Error: Specify config file")
+        sys.exit(0)
+
+    file_name = sys.argv[1]
+    if os.path.exists(file_name):
+        return file_name
+    print("Error: No config file found")
+    sys.exit(0)
+
+def on_failure():
+    """Handles shutdown on error"""
+    for lf in log_files:
+        lf.close()
+    for n in nf_list:
+        try:
+            os.system("sudo pkill " + n)
+        except OSError:
+            pass
+    print("Error occurred. Exiting...")
+    sys.exit(1)
+
+def running_services():
+    """Checks running NFs"""
+    while 1:
+        for pl in procs_list:
+            ret_code = pl.poll()
+            if ret_code is not None:
+                on_failure()
+                break
+            time.sleep(.1)
+            continue
```

Comment:
I think these lines should be updated because the break and continue are redundant. @catherinemeadows please correct me, but I think the sleep should be unindented (outside the for loop) so we can catch the errors between NFs faster?
```suggestion

        time.sleep(.1)
```

On a side note, do you think it's possible to run a timer in this function and allow the user to optionally tell config.py how many seconds to run and die automatically without `^C`? This would help with automating the script.